evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+import pandas as pd
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import import_or_raise
+
+
+class ARIMARegressor(Estimator):
+    """
+    Autoregressive Integrated Moving Average Model.
+    The three parameters (p, d, q) are the AR order, the degree of differencing, and the MA order.
+    More information here: https://www.statsmodels.org/devel/generated/statsmodels.tsa.arima_model.ARIMA.html
+
+    """
+    name = "ARIMA Regressor"
+    hyperparameter_ranges = {
+        "p": Integer(0, 10),
+        "d": Integer(0, 10),
+        "q": Integer(0, 10),
+    }
+    model_family = ModelFamily.ARIMA
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, date_column=None, trend='n', p=1, d=0, q=0,
```

Comment:
I've been thinking about whether expecting dates to be specified in the constructor is a good idea when it comes to `AutoMLSearch` integration. I think it's good! The one downside of this approach is that we can't use a DateTimeFeaturizer in the same pipeline as Arima because the DateTimeFeaturizer will drop the original date column. But I think that's ok since we can pass the date info as the index as well.